### DISCUSSION TITLE
Webpack dev server v3 => v4: publicPath

### DISCUSSION THREAD

**Discussion by PierBJX** [webpack/webpack#15069] [General]
With **WebpackDevServer** _v3_ it was possible to configure the property `publicPath` in the `devServer` options. However, with the _v4_ it is not possible anymore. Therefore, I looked to the breaking changes (into the releases) and I found that on this [page](https://github.com/webpack/webpack-dev-server/releases?page=3):
```
module.exports = {
  // ...
  devServer: {
    dev: {
      publicPath: '/publicPathForDevServe',
    },
  },
};
```
However, I am having this issue:

> [webpack-cli] Invalid options object. Dev Server has been initialized using an options object that does not match the API schema.
>  - options has an unknown property 'dev'. These properties are valid:
>    object { allowedHosts?, bonjour?, client?, compress?, devMiddleware?, headers?, historyApiFallback?, host?, hot?, http2?, https?, ipc?, liveReload?, magicHtml?, onAfterSetupMiddleware?
> , onBeforeSetupMiddleware?, onListening?, open?, port?, proxy?, server?, setupExitSignals?, setupMiddlewares?, static?, watchFiles?, webSocketServer? }

How is it possible to have the same behavior than before?

**Comment by alexander-akait** [comment]
Please use `devMiddleware` instead `dev`

**Reply by PierBJX** [reply]
Thanks, where could I found this information about this breaking changes?

**Reply by alexander-akait** [reply]
https://github.com/webpack/webpack-dev-server/blob/master/migration-v4.md

**Reply by PierBJX** [reply]
Thanks
